pkg/kubelet/disk_manager_test.go

```diff
@@ -146,3 +119,130 @@ func testCache(t *testing.T) {
+		Usage:     10 * mb,
+		Capacity:  100 * mb,
+		Available: 90 * mb,
+	}, nil).Once()
+
+	dm := &realDiskSpaceManager{
+		cadvisor:   mockCadvisor,
+		policy:     policy,
+		cachedInfo: map[string]fsInfo{},
+		frozen:     false,
+	}
+	available, err := dm.isSpaceAvailable("root", 10, dm.cadvisor.RootFsInfo)
+	assert.True(available)
+	assert.NoError(err)
+
+	// Threshold case
+	mockCadvisor = new(cadvisor.Mock)
+	mockCadvisor.On("RootFsInfo").Return(cadvisorApi.FsInfo{
+		Usage:     9 * mb,
+		Capacity:  100 * mb,
+		Available: 9 * mb,
+	}, nil).Once()
+
+	dm = &realDiskSpaceManager{
+		cadvisor:   mockCadvisor,
+		policy:     policy,
+		cachedInfo: map[string]fsInfo{},
+		frozen:     false,
+	}
+	available, err = dm.isSpaceAvailable("root", 10, dm.cadvisor.RootFsInfo)
+	assert.False(available)
+	assert.NoError(err)
+
+	// Frozen case
+	mockCadvisor.On("RootFsInfo").Return(cadvisorApi.FsInfo{
+		Usage:     9 * mb,
+		Capacity:  10 * mb,
+		Available: 500 * mb,
+	}, nil).Once()
+
+	dm = &realDiskSpaceManager{
+		cadvisor:   mockCadvisor,
+		policy:     policy,
+		cachedInfo: map[string]fsInfo{},
+		frozen:     true,
+	}
+	available, err = dm.isSpaceAvailable("root", 10, dm.cadvisor.RootFsInfo)
+	assert.True(available)
+	assert.NoError(err)
+
+	// Capacity error case
+	mockCadvisor = new(cadvisor.Mock)
+	mockCadvisor.On("RootFsInfo").Return(cadvisorApi.FsInfo{
+		Usage:     9 * mb,
+		Capacity:  0,
+		Available: 500 * mb,
+	}, nil).Once()
+
+	dm = &realDiskSpaceManager{
+		cadvisor:   mockCadvisor,
+		policy:     policy,
+		cachedInfo: map[string]fsInfo{},
+		frozen:     false,
+	}
+	available, err = dm.isSpaceAvailable("root", 10, dm.cadvisor.RootFsInfo)
+	assert.True(available)
+	assert.Error(err)
+	assert.Contains(fmt.Sprintf("%s", err), "could not determine capacity")
+
+	// Available error case skipped as v2.FSInfo uses uint64 and this
+	// can not be less than 0
+}
+
+func TestUnfreeze(t *testing.T) {
+	dm := &realDiskSpaceManager{
+		cadvisor:   new(cadvisor.Mock),
+		policy:     testPolicy(),
+		cachedInfo: map[string]fsInfo{},
+		frozen:     true,
+	}
+
+	dm.Unfreeze()
+
+	if dm.frozen {
+		t.Errorf("DiskSpaceManager did not unfreeze: %+v", dm)
+	}
+}
+
+func TestIsDockerDiskSpaceAvailable(t *testing.T) {
+	assert, policy, mockCadvisor := setUp(t)
+	policy.RootFreeDiskMB = 10
+	dm, err := newDiskSpaceManager(mockCadvisor, policy)
+	require.NoError(t, err)
+
+	// 500MB available
+	mockCadvisor.On("DockerImagesFsInfo").Return(cadvisorApi.FsInfo{
+		Usage:     9500 * mb,
+		Capacity:  10000 * mb,
+		Available: 500 * mb,
+	}, nil)
+	ok, err := dm.IsDockerDiskSpaceAvailable()
+	assert.NoError(err)
+	assert.True(ok)
+}
+
+func TestIsRootDiskSpaceAvailable(t *testing.T) {
+	assert, policy, mockCadvisor := setUp(t)
+	policy.RootFreeDiskMB = 10
+	dm, err := newDiskSpaceManager(mockCadvisor, policy)
+	assert.NoError(err)
+
+	// 9MB available
+	mockCadvisor.On("RootFsInfo").Return(cadvisorApi.FsInfo{
```

Comment:
I read this as there is 9MB available, and the RootFreeDiskMB policy needs 10MB.  Should assert be false in this case?